handbook/engineering/continuous_releasability.md

```diff
@@ -0,0 +1,62 @@
+We practice **continuous releasability**. This is a variant of [continous
+delivery](https://en.wikipedia.org/wiki/Continuous_delivery) adapted to a monthly release cadence
+that serves the needs of customers running their own Sourcegraph instances.
+
+### Continuous Releasability Contract
+
+* `master` branch is always releasable
+* Every new feature includes a feature flag
+
+**`master` is always releasable**
+
+Every change merged into `master` must preserve the releasability of `master`. *Releasable* means
+that the revision can be released to customers. In particular,
+
+* **Up-to-date changelog and docs**: The changelog and documentation reflect the current feature set.
+  * Docs must be updated as soon as a feature is merged and enabled.
+  * You can merge a feature without yet updating the docs if it is behind a disabled feature flag.
```

Comment:
What about a feature that isn't necessarily have a feature flag but with explicit configuration option has default "off" value? Should we add feature flag in this case too?

Reply:
This seems like a reasonable exception to the rule. What we care most about is ease of disabling. In this case, to disable the buggy feature, we could just remove the option altogether.

Reply:
All that matters is that we can turn the feature off. Config value is fine if it makes sense for us to allow users to disable the feature in perpetuity.